store/postgres/src/subgraphs.graphql

```diff
@@ -21,8 +21,16 @@ type SubgraphVersion @entity {
 type SubgraphDeployment @entity {
     id: ID! # Subgraph IPFS hash
     manifest: SubgraphManifest!
-    failed: Boolean!
+    failed: Boolean! @deprecated(reason: "Use `health`.")
+    health: Health!
     synced: Boolean!
+
+    "If the subgraph has failed, this is the error caused it"
+    fatalError: SubgraphError
+
+    "Sorted by block number, limited to 1000"
+    nonFatalErrors: [SubgraphError!]!
```

Comment:
It's a little awkward that this is not derived - is that because we couldn't distinguish between fatal and non-fatal errors if we derived this field? It might be better to have a data model where the error just has a `is_fatal` flag and the deployment has an `errors` field.

Reply:
Tbh I hadn't considered deriving this, though I did consider an `is_fatal` flag, but there is a complication, we only want to keep the first 1000 non-fatal errors, but we do want to make sure to expose any fatal error. 

In terms of API, I like having a separate `fatalError` because it expresses that there is at most one fatal error, and offers a straight forward way to query it.